manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -160,19 +174,24 @@ public Long getProcessorsCount(String bridgeId, String customerId) {
         return processorDAO.findByBridgeIdAndCustomerId(bridge.getId(), bridge.getCustomerId(), queryInfo);
     }
 
-    @Transactional
     @Override
     public void deleteProcessor(String bridgeId, String processorId, String customerId) {
-        Processor processor = processorDAO.findByIdBridgeIdAndCustomerId(processorId, bridgeId, customerId);
-
-        ProcessorDefinition processorDefinition = jsonNodeToDefinition(processor.getDefinition());
-        ActionProvider actionProvider = actionProviderFactory.getActionProvider(processorDefinition.getRequestedAction().getType());
-
-        connectorService.deleteConnectorIfNeeded(processorDefinition.getResolvedAction(), processor, actionProvider);
+        List<ConnectorEntity> connectorEntitiesToBeDeleted = updateProcessorConnectorForDeletionOnDB(bridgeId, processorId, customerId);
+        connectorEntitiesToBeDeleted.forEach(c -> {
+            LOGGER.info("Firing deletion event for entity: " + c);
+            eventBus.requestAndForget(Events.CONNECTOR_DELETED_EVENT, c);
```

Comment:
same here

Reply:
I agree with @r00ta comment.

 I feel it would be cleaner for our first implementation to have a `createConnector()` and `deleteConnector` methods on the `ConnectorService` to align with the `ProcessorService` and `BridgeService` interfaces. 
 
 Within the `createConnector` method, if we don't need to create a Connector, it's essentially a null-op. Otherwise we do what's required and fire the event from within the `createConnector` method.

Reply:
I think we should revisit the separation of Services - I don't think that having a separate service per entity is useful at all, especially in features dealing with multiple entity types.